packages/@aws-cdk/aws-codedeploy/lib/lambda/custom-deployment-config.ts

```diff
@@ -0,0 +1,172 @@
+   * - For LINEAR, how frequently additional traffic is shifted
+   * - For CANARY, how long to shift traffic before the full deployment
+   */
+  readonly interval: Duration;
+
+  /**
+   * The verbatim name of the deployment config. Must be unique per account/region.
+   * Other parameters cannot be updated if this name is provided.
+   * @default - automatically generated name
+   */
+  readonly deploymentConfigName?: string;
+}
+
+/**
+ * A custom Deployment Configuration for a Lambda Deployment Group.
+ * @resource AWS::CodeDeploy::DeploymentGroup
+ */
+export class CustomLambdaDeploymentConfig extends Resource implements ILambdaDeploymentConfig {
+
+  /**
+   * The name of the deployment config
+   * @attribute
+   */
+  public readonly deploymentConfigName: string;
+
+  /**
+   * The arn of the deployment config
+   * @attribute
+   */
+  public readonly deploymentConfigArn: string;
+
+  public constructor(scope: Construct, id: string, props: CustomLambdaDeploymentConfigProps) {
+    super(scope, id);
+    this.validateParameters(props);
+
+    // In this section we make the argument for the AWS API call
+    // The name of the config is <construct unique id>.Lambda<deployment type><percentage>Percent<interval>Minutes
+    // Unless the user provides an explicit name
+
+    const deploymentType = 'TimeBased' + props.type.toString();
+    const intervalMinutes = props.interval.toMinutes().toString();
+    const percentage = props.percentage.toString();
+
+    let routingConfig; // The argument to the AWS API call
+    let customDeploymentConfigName = `${percentage}Percent${intervalMinutes}Minutes`;
+    if (props.type == CustomLambdaDeploymentConfigType.CANARY) {
+      routingConfig = {
+        type: deploymentType,
+        timeBasedCanary: {
+          canaryInterval: intervalMinutes,
+          canaryPercentage: percentage,
+        },
+      };
+      customDeploymentConfigName = `LambdaCanary${customDeploymentConfigName}`;
+    } else if (props.type == CustomLambdaDeploymentConfigType.LINEAR) {
+      routingConfig = {
+        type: deploymentType,
+        timeBasedLinear: {
+          linearInterval: intervalMinutes,
+          linearPercentage: percentage,
+        },
+      };
+      customDeploymentConfigName = `LambdaLinear${customDeploymentConfigName}`;
+    }
+
+    // What the physical resource ID is for Cloudformation
+    // Designed so that, if the construct name, percentage, or interval changes,
+    // the deployment config will be deleted and a new one created in its place
+    let resourceName = this.node.uniqueId + '.' + customDeploymentConfigName;
+
+    // If resource name is provided, add it to the physical resource ID so CFN will
+    // update it if the user changes it.
+    if (props.deploymentConfigName !== undefined) {
+      resourceName = props.deploymentConfigName + resourceName;
+    }
+
+    // What you would see in the AWS console
+    const consoleName = props.deploymentConfigName ?? resourceName;
```

Comment:
Sorry, what `consoleName`? There has to be a single physical name, you shouldn't have 2 concepts for it.

Reply:
I agree this is confusing -- see the latest commit to clarify it.